libraries/aws/shadow/include/shadow.h

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file shadow.h
+ * @brief user-facing Shadow functions, and parameter structs.
+ */
+
+#ifndef _SHADOW_H_
+#define _SHADOW_H_
+
+#include <stdint.h>
+/* Include config file before other headers. */
+#include "shadow_config.h"
+
+/*--------------------------- Shadow types ---------------------------*/
+
+/**
+ * @brief Each of these values describe the type of a shadow message.
+ *        https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html
+ */
+typedef enum ShadowMessageType
+{
+    ShadowMessageTypeGetAccepted = 0,
+    ShadowMessageTypeGetRejected,
+    ShadowMessageTypeDeleteAccepted,
+    ShadowMessageTypeDeleteRejected,
+    ShadowMessageTypeUpdateAccepted,
+    ShadowMessageTypeUpdateRejected,
+    ShadowMessageTypeUpdateDocument,
+    ShadowMessageTypeUpdateDelta,
+    ShadowMessageTypeMaxNum
+} ShadowMessageType_t;
+
+/**
+ * @brief Each of these values describe the type of a shadow topic string.
+ *
+ * These are used for topicType parameter of Shadow_GetTopicString() to tell it
+ * what topic string to assemble.
+ */
+typedef enum ShadowTopicStringType
+{
+    /* MISRA Rule 5.2 flags the following member's name as ambiguous from the
```

Comment:
General comment:
We should avoid using coverity exceptions unless it is absolutely necessary. We already have a config file that contains agreed upon MISRA exceptions. For the new library we are trying to avoid added annotations. 5.2 and 5.4 are 'required' rules.

Reply:
This is following the pattern from HTTP
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/http/src/private/http_client_internal.h#L72

Would appreciate to know if we have a better way to handle this case.

SHADOW_TOPIC_STRING_TYPE_GET to ShadowTopicStringTypeGet will be changed in next commit

Reply:
fixed 5.2 with commit [a9b690e](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/a9b690e63f7c1b3fbe90b7328fabb96b7f119eb7)

resolving this comment

feel free to let me know if it needs further changes, thank you